### ISSUE TITLE
Heapster showing error 404 not found

### ISSUE THREAD

**Issue by karunakar1122v** [kubernetes/heapster#1481]
I had created master and 1 node...Kubernetes-dashboard,grafana ,heapster and influxdb running on node.but when I accessing heapster showing error 404 not found.
When I check influxdb URL also not hitting.
Please let me know what was the issue.

**Comment by karunakar1122v** [comment]
https://github.com/kubernetes/heapster/tree/master/deploy/kube-config/influxdb 
I installed using above comfigurations

**Comment by Demonsthere** [comment]
Hi there,
I have the same problem, but from I can see here:
https://console.cloud.google.com/kubernetes/images/tags/heapster-grafana-amd64?location=GLOBAL&project=google-containers
The name of the container changed to heapster-grafana-amd64, but in the yml file there is still gcr.io/google_containers/heapster-grafana:v4.0.2

**Comment by whereisaaron** [comment]
Hi @karunakar1122v, the problem with the current image builds have caught me and a lot of people

1) The [current versions](kubernetes/heapster/tree/master/deploy/kube-config/influxdb) are:
`gcr.io/google_containers/heapster-influxdb-amd64:v1.1.1`
`gcr.io/google_containers/heapster-grafana-amd64:v4.0.2`
`gcr.io/google_containers/heapster:v1.3.0-beta.0`
As you found, due so some image build bug, you need to add the architecture, e.g. `-amd64`.

2) Once you are using influxdb v1.1.1, the image has a bug were it doesn't initialize the correction retention policy, so heapster can't write. I fix that as follows. (See #1482 and #1474).

```
#!/bin/bash

#
# Applies to heapster influxdb image v1.1.1
# https://github.com/kubernetes/heapster/issues/1482
# https://github.com/kubernetes/heapster/issues/1474
#

kubectl run influxcli --restart=Never --rm -it --image=cburki/influxdb-shell -- \
  --host=monitoring-influxdb.kube-system --port=8086 --database=k8s \
  --execute 'CREATE RETENTION POLICY "default" ON k8s DURATION 0d REPLICATION 1 DEFAULT; SHOW RETENTION POLICIES'
```

**Comment by karunakar1122v** [comment]
Thanks for your reply...Will get back to you on this...Once I will try

**Comment by karunakar1122v** [comment]
Still I am facing issue Heapster failed to pushdata to sink:InfluxDb sink.
PFA logs screenshots of Heapster and InfluxDB.

![Uploading heapster.PNG…]()
![Uploading InfluxDb.PNG…]()

**Comment by karunakar1122v** [comment]
Hi Team,

Now I am able to see Graphs in Garfana.
Only issue is when I was working on autoscaling when I execute **kubectl get hpa** command always it showing below error.In below current always showing waiting.
name                        CURRENT     MINPODS   MAXPODS   AGE
student      50%        <waiting>   1         10        46m

**Comment by karunakar1122v** [comment]
Please anyone respond ASAP

**Comment by DirectXMan12** [comment]
That usually means the kubernetes is not able to talk to your copy of Heapster.  Can you please show the events from your HPA object?  Is Heapster deployed in the right namespace?

**Comment by KarolKraskiewicz** [comment]
Configuration files from:
https://github.com/kubernetes/heapster/tree/master/deploy/kube-config/influxdb
have been updated and I verified they are working now.

Not sure how the **hpa** issue is related to heapster.  If hpa is still causing problems, it could be a better idea to open a separate ticket for that.

@karunakar1122v - is there anything else missing here or the ticket can be closed?
